UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -138,13 +137,20 @@ public static Matrix MakeNewMatrix(string Name = "Matrix")
 
 		if (Object == null)
 		{
-			Object = Spawn.ServerPrefab(MatrixPrefab).GameObject;
+			Object = Spawn.ServerPrefab(MatrixPrefab, parent: Instance.transform).GameObject;
```

Comment:
that doesn't sound right.
why are you parenting it to a static manager?

Reply:
because otherwise it will try finding a matrix and it won't find one and then it will proceed error, Because this can initialise by spawning in the space matrix , this is the matrix item as well, it also gets moved in the lines Right after the

Reply:
Parenting a game object under a static manager sounds like a hack that will come to bite us later.
I would try to find a matrix for it to attach to based on where it's located in world position, instead of putting it under something that isn't designed to hold any gameplay objects.

Reply:
it gets repainted instantly

Reply:
reparented*

Reply:
in that case, it's okay.

Reply:
cool

Reply:
I would leave a comment explaining this behavior so if someone wants to grab the matrix of an object that just spawned or if a crowbar is stuck under matrix manager, they can understand why its returning matrix manager.